x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -961,9 +967,14 @@ export class AlertsClient {
     updatedAt: SavedObject['updated_at'] = createdAt,
     references: SavedObjectReference[] | undefined
   ): PartialAlert {
+    const rawAlertWithoutExecutionStatus: Partial<Omit<RawAlert, 'executionStatus'>> = {
+      ...rawAlert,
+    };
+    delete rawAlertWithoutExecutionStatus.executionStatus;
+    const executionStatus = alertExecutionStatusFromRaw(rawAlert.executionStatus);
```

Comment:
this was _sooooo_ hard - I tried multiple approaches to this, including doing what you suggested.  Not happy with that `delete`, and I this seemed like the best alternative.

The main problem is the types of the props in the RawAlert and Alert pretty much match up, so that `...rawAlertWithoutExecutionStatus` fills in most of the bits without any typing errors.  However, the types of the executionStatus in RawAlert and Alert are different.  And because we're dealing with partials coming in and going out, it might not be there.  So when doing something like what you suggest, you'll end up with an error TS thinks the type of the returned executionStatus is RawAlertExecutionStatus | AlertExecutionStatus, which clearly isn't kosher.

It's worth a comment I think, will help the next person looking at this save a few minutes when they try to fix it. heh